zeebe/backup/src/main/java/io/camunda/zeebe/backup/metrics/BackupManagerMetrics.java

```diff
@@ -7,102 +7,112 @@
-  private static final String LIST_OPERATION = "list";
-  private static final String DELETE_OPERATION = "delete";
-
-  private static final Counter TOTAL_OPERATIONS =
-      Counter.build()
-          .namespace(NAMESPACE)
-          .name("backup_operations_total")
-          .help("Total number of backup operations")
-          .labelNames(LABEL_NAME_PARTITION, LABEL_NAME_OPERATION, LABEL_NAME_RESULT)
-          .register();
-  private static final Gauge OPERATIONS_IN_PROGRESS =
-      Gauge.build()
-          .namespace(NAMESPACE)
-          .name("backup_operations_in_progress")
-          .help("Number of backup operations that are in progress")
-          .labelNames(LABEL_NAME_PARTITION, LABEL_NAME_OPERATION)
-          .register();
-
-  private static final Histogram BACKUP_OPERATION_LATENCY =
-      Histogram.build()
-          .namespace(NAMESPACE)
-          .name("backup_operations_latency")
-          .help("Latency of backup operations")
-          .labelNames(LABEL_NAME_PARTITION, LABEL_NAME_OPERATION)
-          .buckets(0.01, 0.1, 1, 10, 60, 5 * 60)
-          .register();
-
-  private final String partitionId;
-
-  public BackupManagerMetrics(final int partitionId) {
-    this.partitionId = String.valueOf(partitionId);
+
+  private final MeterRegistry registry;
+  private final Table<OperationType, OperationResult, Counter> totalOperations = Table.simple();
+  private final EnumMap<OperationType, AtomicLong> operationInProgress =
+      new EnumMap<>(OperationType.class);
+  private final EnumMap<OperationType, Timer> backupOperationLatency =
+      new EnumMap<>(OperationType.class);
+
+  public BackupManagerMetrics(final MeterRegistry meterRegistry) {
+    registry = Objects.requireNonNull(meterRegistry, "meterRegistry cannot be null");
   }
 
   public OperationMetrics startTakingBackup() {
-    return OperationMetrics.start(partitionId, TAKE_OPERATION);
+    return start(OperationType.TAKE);
   }
 
   public OperationMetrics startQueryingStatus() {
-    return OperationMetrics.start(partitionId, STATUS_OPERATION);
+    return start(OperationType.STATUS);
   }
 
   public OperationMetrics startListingBackups() {
-    return OperationMetrics.start(partitionId, LIST_OPERATION);
+    return start(OperationType.LIST);
   }
 
   public OperationMetrics startDeleting() {
-    return OperationMetrics.start(partitionId, DELETE_OPERATION);
+    return start(OperationType.DELETE);
   }
 
   public void cancelInProgressOperations() {
-    OPERATIONS_IN_PROGRESS.labels(partitionId, TAKE_OPERATION).set(0);
-    OPERATIONS_IN_PROGRESS.labels(partitionId, DELETE_OPERATION).set(0);
-    OPERATIONS_IN_PROGRESS.labels(partitionId, STATUS_OPERATION).set(0);
-    OPERATIONS_IN_PROGRESS.labels(partitionId, LIST_OPERATION).set(0);
+    for (final var operation : OperationType.values()) {
+      final var value = operationInProgress.get(operation);
+      if (value != null) {
+        value.set(0L);
+      }
+    }
```

Comment:
:wrench: Can't we use `EnumMeter` for this?

Reply:
In EnumMeter the value is 0 | 1, while here we have counters. Ideally they should be either 0 or 1 as well, but I'm not sure if we should change it